src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -1196,7 +1205,7 @@ private SynthesizedSealedPropertyAccessor MakeSynthesizedSealedAccessor()
         AttributeLocation IAttributeTargetSymbol.DefaultAttributeLocation => AttributeLocation.Property;
 
         AttributeLocation IAttributeTargetSymbol.AllowedAttributeLocations
-            => IsAutoPropertyWithGetAccessor // PROTOTYPE(semi-auto-props): Adjust this and add tests.
+            => HasNonErrorBackingField // PROTOTYPE(semi-auto-props): add tests.
```

Comment:
>HasNonErrorBackingField [](http://example.com/codeflow?start=15&length=23)

Could this be as simple as checking if getter exists and is auto-implemented instead?

Reply:
@AlekseyTs I intended to have an abstraction in place that will work for both auto and semi-auto properties.

Reply:
> I intended to have an abstraction in place that will work for both auto and semi-auto properties.

Then I would go with more specific abstraction, i.e. something like "AllowFieldAttributeTarget" and implement it as suggested.

Reply:
@AlekseyTs Are we fine with "duplicate" implementations for `AllowInitializer`, `AllowFieldAttributeTarget`, and `DisallowRefLikeTypes`?

Reply:
> Are we fine with "duplicate" implementations for ```AllowInitializer```, ```AllowFieldAttributeTarget```, and ```DisallowRefLikeTypes```?

Given that the logic is trivial, I think duplicating the condition is perfectly fine. In exchange we get a flexibility to adjust the logic independently.

Reply:
At the end, we might end up merging some of the properties.